### ISSUE TITLE
Can't use pickled scenegraph objects in parallel worker processes

### ISSUE THREAD

**Issue by jacklovell** [raysect/source#239]
When passing scenegraph objects (such as meshes, or the world) to a worker process using multiprocessing, the objects are passed without error but then cannot be used as normal.

MWE:
```python
from itertools import repeat
from concurrent.futures import ProcessPoolExecutor
from raysect.optical import World
from raysect.primitive.mesh import import_obj
from raysect.optical.library import schott
from raysect.core.scenegraph.utility import print_scenegraph

def worker(bunny):
    print_scenegraph(bunny)

world = World()

#  BUNNY
mesh = import_obj("/home/jlovell/cherab/raysect/demos/resources/stanford_bunny.obj",
                  parent=world,
                  material=schott("N-BK7"))

print("Serial")
worker(mesh)

print("Parallel")
WORKERS = 2
with ProcessPoolExecutor(max_workers=WORKERS) as ex:
    list(ex.map(worker, repeat(mesh, WORKERS)))
```

Run on freia002 with Python 3.5, this gives the following output:
```
jlovell@freia002 ~> python3 /common/scratch/jlovell/raysect_pickle.py 
Serial
<World at 0x7fd96c2509a8>
 |  
 |_ <Mesh at 0x7fd96d1f7f60> [referring node]
Parallel
None
None
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/concurrent/futures/process.py", line 175, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/concurrent/futures/process.py", line 153, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/concurrent/futures/process.py", line 153, in <listcomp>
    return [fn(*args) for args in chunk]
  File "/common/scratch/jlovell/raysect_pickle.py", line 9, in worker
    print_scenegraph(bunny)
  File "raysect/core/scenegraph/utility.pyx", line 103, in raysect.core.scenegraph.utility.print_scenegraph (raysect/core/scenegraph/utility.c:1419)
AttributeError: 'NoneType' object has no attribute 'children'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/common/scratch/jlovell/raysect_pickle.py", line 23, in <module>
    list(ex.map(worker, repeat(mesh, WORKERS)))
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/concurrent/futures/_base.py", line 556, in result_iterator
    yield future.result()
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/concurrent/futures/_base.py", line 405, in result
    return self.__get_result()
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/concurrent/futures/_base.py", line 357, in __get_result
    raise self._exception
AttributeError: 'NoneType' object has no attribute 'children'

```
It seems that the objects are not being properly restored when they're unpickled.

**Comment by jacklovell** [comment]
As suggested, I've updated to Cython 0.28.5 on Freia, and am now getting the following output:
```
jlovell@freia002 ~> python3 /common/scratch/jlovell/raysect_pickle.py 
Serial
<World at 0x7ff5880e69a8>
 |  
 |_ <Mesh at 0x7ff5826b60f0> [referring node]
Parallel
Process Process-1:
Traceback (most recent call last):
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/multiprocessing/process.py", line 254, in _bootstrap
    self.run()
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/concurrent/futures/process.py", line 169, in _process_worker
    call_item = call_queue.get(block=True)
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/multiprocessing/queues.py", line 113, in get
    return ForkingPickler.loads(res)
  File "raysect/optical/material/dielectric.pyx", line 97, in raysect.optical.material.dielectric.Sellmeier.__setstate__
  File "raysect/optical/spectralfunction.pyx", line 340, in raysect.optical.spectralfunction.NumericallyIntegratedSF.__setstate__
AttributeError: 'raysect.optical.material.dielectric.Sellmeier' object has no attribute 'sample_resolutions'
Traceback (most recent call last):
  File "/common/scratch/jlovell/raysect_pickle.py", line 23, in <module>
    list(ex.map(worker, repeat(mesh, WORKERS)))
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/concurrent/futures/_base.py", line 556, in result_iterator
    yield future.result()
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/concurrent/futures/_base.py", line 405, in result
    return self.__get_result()
  File "/usr/local/depot/Python-3.5.1/lib/python3.5/concurrent/futures/_base.py", line 357, in __get_result
    raise self._exception
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
```
It's a different error, at least...

**Comment by CnlPepper** [comment]
Fixed in commit 2b80b5c42a2145af04a5e593e6df2fb77a8a4066
